GoogleMapsComponents/DependencyInjectionExtensions.cs

```diff
@@ -1,50 +1,27 @@
-using System;
-
-namespace GoogleMapsComponents;
-
-public static class DependencyInjectionExtensions
-{
-    internal static BlazorGoogleMapsOptions MapOptions { get; set; } = new BlazorGoogleMapsOptions();
-    public static IServiceCollection AddBlazorGoogleMaps(this IServiceCollection service, Action<BlazorGoogleMapsOptions>? opts = null)
-    {
-        MapOptions = new BlazorGoogleMapsOptions();
-        opts?.Invoke(MapOptions);
-
-        if (MapOptions.UseBootstrapLoader)
-        {
-            if (MapOptions.KeyProvider is null)
-            {
-                throw new ArgumentNullException(nameof(MapOptions.KeyProvider), "Key is required when using bootstrap loader.");
-            }
-
-        }
-
-        return service;
-    }
-}
-
-//All options
-//https://developers.google.com/maps/documentation/javascript/load-maps-js-api#required_parameters
-public class BlazorGoogleMapsOptions
-{
-    /// <summary>
-    /// Is selected option then it uses the bootstrap loader to load the google maps script.
-    /// https://developers.google.com/maps/documentation/javascript/overview#Loading_the_Maps_API
-    /// </summary>
-    public bool UseBootstrapLoader { get; set; }
-
-    /// <summary>
-    /// Default is weekly.
-    /// https://developers.google.com/maps/documentation/javascript/versions
-    /// </summary>
-    public string Version { get; set; } = "weekly";
-
-    /// <summary>
-    /// Comma separated list of libraries to load.
-    /// https://developers.google.com/maps/documentation/javascript/libraries
-    /// </summary>
-    public string? Libraries { get; set; }
-
-    public Func<string>? KeyProvider { get; set; }
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+
+namespace GoogleMapsComponents;
+
+public static class DependencyInjectionExtensions
+{
+    /// <summary>
+    /// Adds a basic service to provide a google api key for the application when using bootstrap loader method for Google API <para/>
+    /// 
+    /// If implementing your own key service, be sure to register the service as the implementation of <see cref="IBlazorGoogleMapsKeyService"/>,
+    /// so that the <see cref="GoogleMap"/> component can find it. For example instead of this function you might use:
+    /// <code>services.AddScoped&lt;<see cref="IBlazorGoogleMapsKeyService"/>, YourServiceImplmentation&gt;();</code>
+    /// </summary>
+    public static IServiceCollection AddBlazorGoogleMaps(this IServiceCollection services, string key)
+    {
+        services.AddScoped<IBlazorGoogleMapsKeyService>(serviceProvider => new BlazorGoogleMapsKeyService(key));
+        return services;
+    }
+
+    /// <inheritdoc cref="AddBlazorGoogleMaps(IServiceCollection, string)"/>
+    public static IServiceCollection AddBlazorGoogleMaps(this IServiceCollection services, Maps.MapApiLoadOptions opts)
+    {
+        services.AddScoped<IBlazorGoogleMapsKeyService>(serviceProvider => new BlazorGoogleMapsKeyService(opts));

```

Comment:
_Client-side doesn't currently have a concept of DI scopes. Scoped-registered services behave like Singleton services_
I hope you tested it with WASM. I agree with your use case when different clients need different api key.

Reply:
I admit I'm not very familiar with the peculiarities of WASM as my company is not using it. 

That said, if the reason that Scoped are the same as Singleton on WASM is that everything is running client side and therefore Singletons would get instantiated at the same time as Scoped anyway (e.g. on browser refresh or forced HTTP request), then I'm not sure what the problem is. As long as WASM still has the `.AddScoped<T>()` overload this shouldn't be and issue right?